### PR TITLE
ICacheObject.put and IAnimateService.addClass Return Types

### PR DESCRIPTION
**Pull request by JohnYoungers**
Updating ICacheObject.put return type to the type of object passed in (opposed to just void):
https://docs.angularjs.org/api/ng/type/$cacheFactory.Cache

Updating IAnimateService.addClass to return the type IPromise<any> (opposed to just void):
https://docs.angularjs.org/api/ngAnimate/service/$animate

### CODE REVIEW

File: angularjs/angular.d.ts [hunk 1]
```diff
@@ -1024,7 +1024,7 @@ declare module ng {
             // Not garanteed to have, since it's a non-mandatory option
             //capacity: number;
         };
-        put(key: string, value?: any): void;
+        put<T>(key: string, value?: T): T;
```
Comment (vvakame) [inline angularjs/angular.d.ts]: umm...

```
var cache; ng. ICacheObject;
cache.put("test", true); // return undefined
cache.put("test", 1); // return true
```

I think `put(key: string, value?: any): any;` is better.

Comment (jayoungers) [inline angularjs/angular.d.ts]: How did you go about getting those results?  When I run those examples I get the values back as expected:
http://plnkr.co/edit/VI1d3uyyRIPTI0rooNVO?p=preview

Comment (vvakame) [inline angularjs/angular.d.ts]: oh, sorry. I was read a document. Returns is `the value stored.`. but my comprehension is invalid.

**Comment by vvakame** [comment]
@jayoungers thanks mate!

**Pull request merged**